owncloudApp/src/main/java/com/owncloud/android/MainApp.kt

```diff
@@ -159,9 +159,17 @@ class MainApp : Application() {
         initDependencyInjection()
     }
 
-    fun startLogsIfEnabled() {
-        enabledLogging = PreferenceManager.getDefaultSharedPreferences(applicationContext)
-            .getBoolean(SettingsLogsFragment.PREFERENCE_ENABLE_LOGGING, false)
+    private fun startLogsIfEnabled() {
+        val preferenceManager = PreferenceManager.getDefaultSharedPreferences(applicationContext)
+
+        if (BuildConfig.DEBUG) {
+            val alreadySet = preferenceManager.contains(PREFERENCE_ENABLE_LOGGING)
+            if (!alreadySet) {
+                preferenceManager.edit().putBoolean(PREFERENCE_ENABLE_LOGGING, true).apply()
```

Comment:
Maybe it's better to use `SharedPreferencesProviderImpl` so that we always use this interface in a consistent way, and with no need to call `apply()` nor `edit()`. What do you think? 🤔